lib/yard/parser/c/c_parser.rb

```diff
@@ -38,6 +38,7 @@ def parse_toplevel
             when '#'; consume_directive
             when '/'; consume_comment
             when /\s/; consume_whitespace
+            when '}'; advance # Skip possible C++ namespace closing brackets.
```

Comment:
I'm not too concerned, since the worst case is that it continues past a toplevel element to find the next one.